update_existing_families.py

```diff
@@ -170,16 +170,30 @@ def save_as_csv_file(filename, header_fields, data):
         #   (ii) this student is returning after a break from BSD, but somehow was given a new student ID
         # If either (i) or (ii) is true, then DP will not create a new record, but update the existing donor record, so we should be okay. 
         # At any rate, we have to prepare a new record for this donor, with several custom fields
-        print("Creating record for new donor w/ new student %s" % (student_id))
+        #print("Creating record for new donor w/ new student %s" % (student_id))
 
-        h_f_name = district_record['Parent 1 First Name']
-        w_f_name = district_record['Parent 2 First Name']
         h_l_name = district_record['Parent1 Last Name']
-        w_l_name = district_record['Parent 2 Last Name']
-        if w_l_name == h_l_name :
-            salutation = h_f_name + " and " + w_f_name + " " + w_l_name
+        if len(h_l_name) != 0: 
+            h_f_name = district_record['Parent 1 First Name']
+            w_f_name = district_record['Parent 2 First Name']
+            w_l_name = district_record['Parent 2 Last Name']
         else:
-            salutation = h_f_name + " " + h_l_name + " and " + w_f_name + " " + w_l_name
+            h_f_name = district_record['Parent 2 First Name']
+            h_l_name = district_record['Parent 2 Last Name']
+            w_f_name = ""
+            w_l_name = ""
+
+
+        if len(w_l_name) != 0:
+            if w_l_name == h_l_name :
+                salutation = h_f_name + " and " + w_f_name + " " + w_l_name
+            else:
+                salutation = h_f_name + " " + h_l_name + " and " + w_f_name + " " + w_l_name
+            informal_sal = h_f_name + " and " + w_f_name
+        else:
+            salutation = h_f_name + " " + h_l_name 
+            informal_sal = h_f_name 
+
```

Comment:
I think there are still bugs here. E.g. if there is only parent 2, you're not assigning them as the only parent. I think you want to break up 3 cases (only parent1, only parent2, both), and assign salutation, informal salutation, and both sets of first and last names in each case.